apps/emqx_connector/test/emqx_connector_pgsql_SUITE.erl

```diff
@@ -0,0 +1,122 @@
+% %% you may not use this file except in compliance with the License.
+% %% You may obtain a copy of the License at
+% %% http://www.apache.org/licenses/LICENSE-2.0
+% %%
+% %% Unless required by applicable law or agreed to in writing, software
+% %% distributed under the License is distributed on an "AS IS" BASIS,
+% %% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% %% See the License for the specific language governing permissions and
+% %% limitations under the License.
+% %%--------------------------------------------------------------------
+
+-module(emqx_connector_pgsql_SUITE).
+
+-compile(nowarn_export_all).
+-compile(export_all).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("emqx/include/emqx.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-define(PGSQL_HOST, "pgsql").
+-define(PGSQL_PORT, 5432).
+
+all() ->
+    emqx_common_test_helpers:all(?MODULE).
+
+groups() ->
+    [].
+
+init_per_suite(Config) ->
+    case emqx_common_test_helpers:is_tcp_server_available(?PGSQL_HOST, ?PGSQL_PORT) of
+        true ->
+            Config;
+        false ->
+            {skip, no_pgsql}
+    end.
+
+end_per_suite(_Config) ->
+    ok.
+
+% %%------------------------------------------------------------------------------
+% %% Testcases
+% %%------------------------------------------------------------------------------
+
+% Simple test to make sure the proper reference to the module is returned.
+t_roots(_Config) ->
+    ExpectedRoots = [{config,#{type => {ref,emqx_connector_pgsql,config}}}],
+    ActualRoots = emqx_connector_pgsql:roots(),
+    ?assertEqual(ExpectedRoots, ActualRoots).
+
+% Not sure if this level of testing is appropriate for this function.
+% Checking the actual values/types of the returned term starts getting
+% into checking the emqx_connector_schema_lib.erl returns and the shape
+% of expected data elsewhere.
+t_fields(_Config) ->
+    Fields = emqx_connector_pgsql:fields(config),
+    lists:foreach(
+        fun({FieldName, FieldValue}) ->
+            ?assert(is_atom(FieldName)),
+            if
+                is_map(FieldValue) ->
+                    ?assert(maps:is_key(type, FieldValue) and maps:is_key(default, FieldValue));
+                true ->
+                    ?assert(is_function(FieldValue))
+            end
+        end,
+        Fields).
+
+% Simple start and stop of the connector
+t_start_and_stop_successfully(_Config) ->
+    ?assertEqual({ok, #{poolname => emqx_connector_pgsql}}, emqx_connector_pgsql:on_start(<<"emqx_connector_pgsql">>, pgsql_config())),
+    ?assertEqual(ok, emqx_connector_pgsql:on_stop(<<"emqx_connector_pgsql">>, #{poolname => emqx_connector_pgsql})).
+
+% Start the connector, execute a minimal query, and then stop it again.
+t_start_query_and_stop_successfully(_Config) ->
+    ?assertEqual({ok, #{poolname => emqx_connector_pgsql}}, emqx_connector_pgsql:on_start(<<"emqx_connector_pgsql">>, pgsql_config())),
```

Comment:
I suggest starting and stopping the connection in `init_per_testcase` and `end_per_testcase`, respectively.  This ensures that the service is stopped even if some assertions fail, and also makes the test body slimmer and easier to read.